pymc_experimental/tests/test_model_builder.py

```diff
@@ -122,6 +131,38 @@ def default_sampler_config(self) -> Dict:
         }
 
 
+def test_save_input_params(fitted_model_instance):
+    assert fitted_model_instance.idata.attrs["test_paramter"] == '"test_paramter"'
+
+
+def test_save_load(fitted_model_instance):
+    temp = tempfile.NamedTemporaryFile(mode="w", encoding="utf-8", delete=False)
+    fitted_model_instance.save(temp.name)
+    test_builder2 = test_ModelBuilder.load(temp.name)
+    assert fitted_model_instance.idata.groups() == test_builder2.idata.groups()
+    assert fitted_model_instance.id == test_builder2.id
+    x_pred = np.random.uniform(low=0, high=1, size=100)
+    prediction_data = pd.DataFrame({"input": x_pred})
+    pred1 = fitted_model_instance.predict(prediction_data["input"])
+    pred2 = test_builder2.predict(prediction_data["input"])
+    assert pred1.shape == pred2.shape
+    temp.close()
+
+
+def test_convert_dims_to_tuple(fitted_model_instance):
+    model_config = {
+        "a": {
+            "loc": 0,
+            "scale": 10,
+            "dims": [
+                "x",
+            ],
+        },
+    }
+    converted_model_config = fitted_model_instance._convert_dims_to_tuple(model_config)
+    assert converted_model_config["a"]["dims"] == ("x",)
```

Comment:
Would be more useful to have a test for the "global functionality" and not the specific function used internally. My understanding is that models with dims failed to save and load, and now they work. Can you test along those lines?

Then the test will stay relevant even if the internal way that's achieved changes.

Reply:
Each save/load test in the child classes test this fix, as load itself checks the model IDs which are dependent on model config content and some other things

Reply:
I don't see why any of the other tests would fail if we reverted the fix. Only this test seems to actually check for dims?

Reply:
Good catch, for some reason I omitted the id comparison in load while implementing load function in clv models. That's why it didn't complain about it the way DelayedSaturatedMMM was. Now it's fixed, also addressing the removal of temporary file that was mentioned in marketing

Reply:
you can find the adapted tests here:
[https://github.com/pymc-labs/pymc-marketing/pull/317/commits/99568344d7ba68a54aad7119349f924b10ad4778](modified load with save/load_tests)

Reply:
I made the `_convert_dims_to_tuple` be a No-op, and none of the pre-existing tests failed: 
![image](https://github.com/pymc-devs/pymc-experimental/assets/28983449/93024824-cd12-49fe-9d53-f7c08c69e6ff)

We should include a direct test that dims have the correct type after save/load, in this library :)

Reply:
Made that method simply return the `config_dict` without the logic to convert to tuple

Reply:
@ricardoV94 I've introduced coords into the test model, now if you'd remove the _convert_dims_to_tuple, the load would fail as change from tuple to array in dims would result in different sha256, and in result fail the load id assertion